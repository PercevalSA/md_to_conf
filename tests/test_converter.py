import pytest
from md_to_conf import MarkdownConverter


@pytest.fixture
def test_converter() -> MarkdownConverter:
    return MarkdownConverter(
        "tests/testfiles/basic.md", "https://domain.confluence.net/wiki", "default", 2
    )


def test_converter_init():
    md_file = "tests/testfiles/basic.md"
    url = "https://domain.confluence.net/wiki"
    md_source = "default"
    editor_version = 2

    converter = MarkdownConverter(md_file, url, md_source, editor_version)

    assert converter.md_file == md_file
    assert converter.api_url == url
    assert converter.md_source == md_source
    assert converter.editor_version == editor_version


def test_converter_basic(test_converter: MarkdownConverter, snapshot):
    html = test_converter.get_html_from_markdown()
    assert html == snapshot


def test_slug(test_converter: MarkdownConverter):
    slug = test_converter.slug("<tag>The $slug</tag>", False)
    assert slug == "The-slug"


def test_slug_lower(test_converter: MarkdownConverter):
    slug = test_converter.slug("<tag>The $slug</tag>", True)
    assert slug == "the-slug"


def test_convert_comment_block(test_converter: MarkdownConverter):
    slug = test_converter.convert_comment_block("<!-- some comments go here -->")
    assert slug == "<ac:placeholder> some comments go here </ac:placeholder>"


def test_toc_convert(test_converter: MarkdownConverter, snapshot):
    slug = test_converter.create_table_of_content(
        "<h1>First heading</h1> <p>[TOC]</p> <h2>Second heading</h2>"
    )

    assert slug == snapshot
