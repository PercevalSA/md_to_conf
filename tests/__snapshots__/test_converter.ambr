# serializer version: 1
# name: test_converter_basic
  '''
  <h1>Heading 1</h1>
  <h2>Heading 2</h2>
  <p>Some text for heading 2</p>
  <h3>Heading 3-1</h3>
  <p>Some text for heading 3-1</p>
  <h3>Heading 3-2</h3>
  <p>Some test for heading 3-2</p>
  <ul>
  <li>An</li>
  <li>unordered</li>
  <li>
  <p>list</p>
  </li>
  <li>
  <p>An</p>
  </li>
  <li>Ordered</li>
  <li>List</li>
  </ul>
  <pre><code class="language-less">&gt; This is an info
  </code></pre>
  <pre><code class="language-less">&gt; Note: This is a note
  </code></pre>
  <pre><code class="language-less">&gt; Warning: This is a warning
  </code></pre>
  '''
# ---
# name: test_toc_convert
  '<h1>First heading</h1> <p><ac:structured-macro ac:name="toc" ac:schema-version="1"/></p> <h2>Second heading</h2>'
# ---
